java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueViewVector.java

```diff
@@ -0,0 +1,327 @@
+    this(name, allocator, DEFAULT_DATA_VECTOR, callBack);
+  }
+
+  protected BaseRepeatedValueViewVector(
+      String name, BufferAllocator allocator, FieldVector vector, CallBack callBack) {
+    super(allocator);
+    this.name = name;
+    this.offsetBuffer = allocator.getEmpty();
+    this.sizeBuffer = allocator.getEmpty();
+    this.vector = Preconditions.checkNotNull(vector, "data vector cannot be null");
+    this.repeatedCallBack = callBack;
+    this.valueCount = 0;
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  @Override
+  public boolean allocateNewSafe() {
+    boolean dataAlloc = false;
+    try {
+      offsetBuffer = allocateOffsetBuffer(offsetAllocationSizeInBytes);
+      sizeBuffer = allocateSizeBuffer(sizeAllocationSizeInBytes);
+      dataAlloc = vector.allocateNewSafe();
+    } catch (Exception e) {
+      e.printStackTrace();
+      clear();
+      return false;
+    } finally {
+      if (!dataAlloc) {
+        clear();
+      }
+    }
+    return dataAlloc;
+  }
+
+  protected ArrowBuf allocateOffsetBuffer(final long size) {
+    final int curSize = (int) size;
+    ArrowBuf offsetBuffer = allocator.buffer(curSize);
+    offsetBuffer.readerIndex(0);
+    offsetAllocationSizeInBytes = curSize;
+    offsetBuffer.setZero(0, offsetBuffer.capacity());
+    return offsetBuffer;
+  }
+
+  protected ArrowBuf allocateSizeBuffer(final long size) {
+    final int curSize = (int) size;
+    ArrowBuf sizeBuffer = allocator.buffer(curSize);
+    sizeBuffer.readerIndex(0);
+    sizeAllocationSizeInBytes = curSize;
+    sizeBuffer.setZero(0, sizeBuffer.capacity());
+    return sizeBuffer;
+  }
+
+  @Override
+  public void reAlloc() {
+    reallocOffsetBuffer();
+    reallocSizeBuffer();
+    vector.reAlloc();
+  }
+
+  protected void reallocOffsetBuffer() {
+    final long currentBufferCapacity = offsetBuffer.capacity();
+    long newAllocationSize = currentBufferCapacity * 2;
+    if (newAllocationSize == 0) {
+      if (offsetAllocationSizeInBytes > 0) {
+        newAllocationSize = offsetAllocationSizeInBytes;
+      } else {
+        newAllocationSize = INITIAL_VALUE_ALLOCATION * OFFSET_WIDTH * 2;
+      }
+    }
+
+    newAllocationSize = CommonUtil.nextPowerOfTwo(newAllocationSize);
+    newAllocationSize = Math.min(newAllocationSize, (long) OFFSET_WIDTH * Integer.MAX_VALUE);
+    assert newAllocationSize >= 1;
+
+    if (newAllocationSize > MAX_ALLOCATION_SIZE || newAllocationSize <= offsetBuffer.capacity()) {
+      throw new OversizedAllocationException("Unable to expand the buffer");
+    }
+
+    final ArrowBuf newBuf = allocator.buffer(newAllocationSize);
+    newBuf.setBytes(0, offsetBuffer, 0, currentBufferCapacity);
+    newBuf.setZero(currentBufferCapacity, newBuf.capacity() - currentBufferCapacity);
+    offsetBuffer.getReferenceManager().release(1);
+    offsetBuffer = newBuf;
+    offsetAllocationSizeInBytes = newAllocationSize;
+  }
+
+  protected void reallocSizeBuffer() {
```

Comment:
Ditto here. It might be better to have a generic "allocate"/"reallocate" helper

Reply:
added "allocate" and "reallocate" helpers